### PR TITLE
Add credit card

### LINKED ISSUE #8: Add Credit Card Information
Using the terminal, a user should be able to add a billing address, card number, expiration date, CVV, and card nickname (default is last 4 digits of card) associated with a credit card so that they can view it later.

### PR DESCRIPTION
**Pull request by mbendy**
Don' t merge yet, just so y'all can view my progress

### CODE REVIEW

**Comment by mbendy** [comment]
@ryanloutos , check out how I made a static method for user prompts to add the card, i think this could be good so that our main in SafeStore.java is less cluttered. However, we have to figure out how to test this, I have an idea though, will work on it later. You may want to do something similar.

File: src/card/Address.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+	public boolean updateToExistingAddress(ConnectionSource databaseConnection) {
+		if (this.addressExists(databaseConnection)) {
+			AddressEntity updatedEntity = this.getExistingEntity(databaseConnection);
+			this.addressEntity = updatedEntity;
+			return true;
+		}
+		return false;
+	}
+	
+	public boolean addAddress(ConnectionSource databaseConnection) {
```
Comment (ryanloutos-wustl) [inline src/card/Address.java:153]: Is there a reason to return a boolean here? If the the insert doesn't work, won't a SQLException just be thrown anyway? What are the other possible return values of create besides 1?

File: test/cardTests/creditCardTests.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+		try {
+			List<AddressEntity> returnedAddresses = addressDao.queryForFieldValues(addressQueryParams);
+			String errMsg = "Only one address should exist in the table with this particular street address, city, state, and zip code";
+			assertTrue(returnedAddresses.size() == 1, errMsg);
+			
+			AddressEntity returnedAddressEntity = returnedAddresses.get(0);
+			ForeignCollection<CreditCardEntity> associatedCreditCards = returnedAddressEntity.getCreditCards();
+			assertTrue(associatedCreditCards.size() == 1, "Address should only have one card associated with it");
+			
+			Dao<CreditCardEntity, String> creditCardDao = DaoManager.createDao(databaseConnection, CreditCardEntity.class);
```
Comment (ryanloutos-wustl) [inline test/cardTests/creditCardTests.java:103]: Might be worth checking to see username is stored correctly so there exists a relationship

File: test/cardTests/creditCardTests.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+			List<AddressEntity> returnedAddresses = addressDao.queryForFieldValues(addressQueryParams);
+			String errMsg = "Only one address should exist in the table with this particular street address, city, state, and zip code";
+			assertTrue(returnedAddresses.size() == 1, errMsg);
+			
+			AddressEntity returnedAddressEntity = returnedAddresses.get(0);
+			ForeignCollection<CreditCardEntity> associatedCreditCards = returnedAddressEntity.getCreditCards();
+			System.out.println("LOOK BOI: " + associatedCreditCards.size());
+			assertTrue(associatedCreditCards.size() == 1, "Address should only have one card associated with it");
+			
+			Dao<CreditCardEntity, String> creditCardDao = DaoManager.createDao(databaseConnection, CreditCardEntity.class);
```
Comment (ryanloutos-wustl) [inline test/cardTests/creditCardTests.java:145]: Same as above

**Review by ryanloutos-wustl — CHANGES_REQUESTED** [review]
Trying to submit a review here but don't know how this works

File: src/card/Address.java (hunk 1 above)
Comment (mbendy) [reply src/card/Address.java:153]: it catches the exception so it wouldn't get thrown, this way when we add an address if it fails bc of a SQLException we can at least alert the user that it failed

**Review by ryanloutos-wustl — APPROVED** [review]

**Pull request merged**